DataFormats/Phase2TrackerDigi/src/Phase2ITChip.cc

```diff
@@ -0,0 +1,130 @@
+#include <vector>
+#include <utility>
+#include <string>
+#include <iostream>
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2ITQCore.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2ITChip.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2ITDigiHit.h"
+
+Phase2ITChip::Phase2ITChip(int rocnum, std::vector<Phase2ITDigiHit> hl) {
+  hitList = hl;
+  rocnum_ = rocnum;
+}
+
+unsigned int Phase2ITChip::size() { return hitList.size(); }
+
+//Returns the position (row,col) of the 4x4 QCores that contains a hit
+std::pair<int, int> Phase2ITChip::get_QCore_pos(Phase2ITDigiHit hit) {
+  int row = hit.row() / 4;
+  int col = hit.col() / 4;
+  return {row, col};
+}
+
+//Takes a hit and returns the 4x4 QCore that contains it
+Phase2ITQCore Phase2ITChip::get_QCore_from_hit(Phase2ITDigiHit pixel) {
+  std::vector<int> adcs(16, 0), hits(16, 0);
+  std::pair<int, int> pos = get_QCore_pos(pixel);
+
+  for (const auto& hit : hitList) {
+    if (get_QCore_pos(hit) == pos) {
+      int i = (4 * (hit.row() % 4) + (hit.col() % 4) + 8) % 16;
+      adcs[i] = hit.adc();
+      hits[i] = 1;
+    }
+  }
+
+  Phase2ITQCore qcore(0, pos.second, pos.first, false, false, adcs, hits);
+  return qcore;
+}
+
+//Removes duplicated Phase2ITQCores
+std::vector<Phase2ITQCore> Phase2ITChip::rem_duplicates(std::vector<Phase2ITQCore> qcores) {
```

Comment:
Hi, for this I'll take note and fix it in the next PR. This is going to be a tentative innertracker chip data processor so that we can first do the DAQ studies